src/Main.cpp

```diff
@@ -68,39 +82,50 @@ class BittyBuzzTask : public AbstractTask<10 * configMINIMAL_STACK_SIZE> {
-            m_logger.log(LogLevel::Error, "BBZVM failed to initialize. state: %s err: %s",
-                         BittyBuzzSystem::getStateString(m_bittybuzzVm.getState()),
-                         BittyBuzzSystem::getErrorString(m_bittybuzzVm.getError()));
-            m_deviceStateUI.setDeviceState(DeviceState::Error);
-            return;
-        }
+        // register btn function
+        m_buttonCallbackRegister.setCallback(resetVmButtonCallback, this);
 
         while (true) {
-            BBVMRet statusCode = m_bittybuzzVm.step();
-            switch (statusCode) {
-            case BBVMRet::Ok:
-                break;
-            case BBVMRet::OutMsgErr: {
-                m_logger.log(LogLevel::Warn, "Buzz could not sent message");
-                m_deviceStateUI.setDeviceState(DeviceState::Error);
-                break;
-            }
-            case BBVMRet::VmErr: {
-                m_logger.log(LogLevel::Error, "BBZVM failed to step. state: %s err: %s",
+            m_deviceStateUI.setDeviceState(DeviceState::Ok);
+
+            if (!m_bittybuzzVm.init(buzzLibraries.data(), buzzLibraries.size()) ||
+                !m_bittybuzzVm.start()) {
+
+                m_logger.log(LogLevel::Error, "",
                              BittyBuzzSystem::getStateString(m_bittybuzzVm.getState()),
                              BittyBuzzSystem::getErrorString(m_bittybuzzVm.getError()));
-                m_deviceStateUI.setDeviceState(vmErrorToDeviceState(m_bittybuzzVm.getError()));
-                break;
-            }
-            default: {
-                m_logger.log(LogLevel::Warn, "Unkown bbzvm step status code");
                 m_deviceStateUI.setDeviceState(DeviceState::Error);
+                return;
             }
+
+            while (!m_resetVm) {
+                BBVMRet statusCode = m_bittybuzzVm.step();
+                switch (statusCode) {
+                case BBVMRet::Ok:
+                    break;
+                case BBVMRet::Stopped: {
+                    m_logger.log(LogLevel::Warn, "BBZVM is stopped, cannot step");
+                    m_deviceStateUI.setDeviceState(DeviceState::Error);
+                    break;
+                }
+                case BBVMRet::OutMsgErr: {
+                    m_logger.log(LogLevel::Warn, "Buzz could not sent message");
+                    m_deviceStateUI.setDeviceState(DeviceState::Error);
+                    break;
+                }
+                case BBVMRet::VmErr: {
+                    m_logger.log(LogLevel::Error, "BBZVM failed to step. state: %s err: %s",
+                                 BittyBuzzSystem::getStateString(m_bittybuzzVm.getState()),
+                                 BittyBuzzSystem::getErrorString(m_bittybuzzVm.getError()));
+                    m_deviceStateUI.setDeviceState(vmErrorToDeviceState(m_bittybuzzVm.getError()));
+                    break;
+                }
+                }
+                Task::delay(100);
```

Comment:
Should this be in a constant somewhere? It seems like something we might want to let the user configure

Reply:
It's in the next PR don't worry. And the VM will stop printing endlessy when it cashes